solutionbox/inception/datalab_solutions/inception/_util.py

```diff
@@ -146,3 +146,56 @@ def accuracy(logits, labels):
   accuracy_op = tf.cast(correct_count, tf.float32) / tf.cast(
       correct_count + incorrect_count, tf.float32)
   return [correct_count_update, incorrect_count_update], accuracy_op
+
+
+def get_sources_from_dataset(p, dataset, mode):
+  """get pcollection from dataset."""
+  import apache_beam as beam
+  import csv
+  from datalab.mlalpha import CsvDataSet, BigQueryDataSet
+
+  if type(dataset) is CsvDataSet:
+    source_list = []
+    for ii, input_path in enumerate(dataset.files):
+      source_list.append(p | 'Read from Csv %d (%s)' % (ii, mode) >> 
+          beam.io.ReadFromText(input_path, strip_trailing_newlines=True))
+    return (source_list | 'Flatten Sources (%s)' % mode >> beam.Flatten()
+        | 'Create Dict from Csv (%s)' % mode >>
+          beam.Map(lambda line: csv.DictReader([line], fieldnames=['image_url', 'label']).next()))
+  elif type(dataset) is BigQueryDataSet:
+    if len(dataset.sql.split()) == 1:
```

Comment:
We should refactor the BigQueryDataSet so it knows if it has a table or sql statement.